### PR TITLE
Add metrics-server to openshift-metrics playbook

### PR DESCRIPTION
**Pull request by frobware**
This commit reintroduces the metric server. It was reverted in 3.10
as we did not have OCP images ready;  it reverts commit 36227617f5ce5e87e77dbc9cd394fe229c62cfd4.

It additionally removes openshift_metrics_server_image_version and
openshift_metrics_server_image_prefix to simpler override of
openshift_metrics_server_image. This will allow users to utilise
oreg_url for this component by default.

It also changes the metrics-server namespace from `openshift-infra` to
`openshift-monitoring`.

### CODE REVIEW

**Comment by frobware** [comment]
Although this is for 3.11 I have tested the changes against an OCP 3.10 deployment:

```console
$ kubectl top nodes
NAME                           CPU(cores)   CPU%      MEMORY(bytes)   MEMORY%
ip-172-18-0-121.ec2.internal   63m          3%        1626Mi          21%
ip-172-18-14-66.ec2.internal   1100m        55%       2774Mi          35%
ip-172-18-4-170.ec2.internal   1278m        63%       3025Mi          39%

$ kubectl top pods --all-namespaces
NAMESPACE               NAME                                              CPU(cores)   MEMORY(bytes)   
default                 docker-registry-2-6gf8w                           1m           8Mi             
default                 router-3-bb2kx                                    3m           34Mi            
kube-system             master-api-ip-172-18-4-170.ec2.internal           42m          756Mi           
kube-system             master-controllers-ip-172-18-4-170.ec2.internal   52m          188Mi           
kube-system             master-etcd-ip-172-18-4-170.ec2.internal          18m          192Mi           
openshift-infra         bootstrap-autoapprover-0                          0m           20Mi            
openshift-infra         hawkular-cassandra-1-qvfl2                        598m         1190Mi          
openshift-infra         hawkular-metrics-bhwdk                            418m         843Mi           
openshift-infra         heapster-w4m5t                                    8m           20Mi            
openshift-monitoring    metrics-server-765d8598db-vsmzw                   0m           10Mi            
openshift-node          sync-gbghx                                        0m           0Mi             
openshift-node          sync-np8km                                        0m           0Mi             
openshift-node          sync-qllnl                                        0m           12Mi            
openshift-sdn           ovs-2cm27                                         1m           33Mi            
openshift-sdn           ovs-9clrf                                         1m           33Mi            
openshift-sdn           ovs-gvhcx                                         1m           33Mi            
openshift-sdn           sdn-4nmj2                                         4m           30Mi            
openshift-sdn           sdn-g27hr                                         4m           32Mi            
openshift-sdn           sdn-n4bgw                                         5m           27Mi            
openshift-web-console   webconsole-99fb9d66f-zl5sm                        17m          11Mi            
```

```console 
$ oc version
oc v3.10.2
kubernetes v1.10.0+b81c8f8
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://ip-172-18-4-170.ec2.internal:8443
openshift v3.10.2
kubernetes v1.10.0+b81c8f8
```

The metrics server is being pulled from `registry.reg-aws.openshift.com/openshift3/ose-metrics-server:v3.10.2`.

**Comment by frobware** [comment]
@DirectXMan12 PTAL to make sure that things haven't moved on since we last did this.

**Comment by vrutkovs** [comment]
bot, retest this please

**Comment by DirectXMan12** [comment]
@jcantrill can you approve this please?

File: roles/openshift_metrics/templates/metrics-server-deployment.j2 [hunk 1]
```diff
@@ -0,0 +1,45 @@
+apiVersion: extensions/v1beta1
```
Comment (derekwaynecarr) [inline roles/openshift_metrics/templates/metrics-server-deployment.j2]: is there a reason we cannot just use the v1 API form?

**Pull request merged**